### PR TITLE
Fix cell and virial transpose bug in dp_ipi

### PR DESCRIPTION
**Pull request by njzjz**
i-Pi assumes column-major storage to store cell and virial.

Fix #1351.
See i-Pi manual: https://ipi-code.org/i-pi/_static/manual.pdf (page 33 and page 78)
See also i-pi/i-pi#48 and i-pi/i-pi#196.

### CODE REVIEW

**Comment by njzjz** [comment]
I am not sure I am correct - @wanghan-iapcm can you take a look?

**Comment by njzjz** [comment]
Also, is #1123 related?

File: source/ipi/driver.cc [hunk 1]
```diff
@@ -158,7 +158,7 @@ int main(int argc, char * argv[])
       readbuffer_ (&socket, (char *)(cell_h),  9*sizeof(double));
       readbuffer_ (&socket, (char *)(cell_ih), 9*sizeof(double));
       for (int dd = 0; dd < 9; ++dd){
-	dbox[dd] = cell_h[dd] * cvt_len;
+	dbox[dd] = cell_ih[dd] * cvt_len;
```
[the remark this answers is not in the document]
Comment (njzjz) [reply source/ipi/driver.cc:161]: Oh the manual said it is the inverse matrix... I didn't realize the difference between them.

> The serversocket will then send a string “POSDATA”, then nine floats for the cell vectormatrix, then another nine floats for the inverse matrix. 
(page 33)

**Comment by njzjz** [comment]
For the system in #1351, the energy error before this PR is 0.02, and the error after this PR is 6e-6.

**Comment by njzjz** [comment]
The rest error should come from the unit conversion, I think.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**